backend/app/controllers/spree/admin/promotion_codes_controller.rb

```diff
@@ -23,7 +23,12 @@ def index
 
       def new
         @promotion = Spree::Promotion.accessible_by(current_ability, :read).find(params[:promotion_id])
-        @promotion_code = @promotion.promotion_codes.build
+        if @promotion.apply_automatically
```

Comment:
What about also performing this check on the create? I know we have the model validation there but I think it's quite useless to build promotion codes if we already know they are not valid. Additionally, we already have this code in the same controller that could easily become a `before_action` for both. What do you think?

Reply:
Yep you're absolutely right, I think is the best way